### PR TITLE
ENG3-1113: Document supported extension hooks

### LINKED ISSUE #49: [Feature] Add documentation for available filters
**Description**
We do not currently have the any documentation explaining the available filters that users can hook into to modify cache behavior.

### PR DESCRIPTION
**Pull request by jacobd91**
https://imh-internal.atlassian.net/browse/ENG3-1113

## Summary
- generate a deterministic reference for 240 first-party W3TC actions and filters
- document accepted callback arguments, registration APIs, and source locations
- add PHP 7.4/8+ parser fixtures and CI drift validation
- link the hook reference from the README

## Test plan
- [x] Generator self-test and freshness check
- [x] Full PHP lint
- [x] PHPCS for the generator
- [x] Full branch diff whitespace check
- [x] Human documentation review

Fixes #49

---

> [!NOTE]
> **Low Risk**
> Documentation and build-time tooling only; no runtime plugin behavior changes.
> 
> **Overview**
> Adds **auto-generated developer documentation** for W3 Total Cache actions and filters by introducing `bin/generate-hooks-docs.php`, which tokenizes production PHP, finds `apply_filters` / `do_action` (and W3TC early boot variants), normalizes dynamic hook names, and writes **`docs/hooks.md`** with registration APIs, accepted argument counts, parameters, and source links.
> 
> **CI and workflow:** `package.json` gains `hooks:generate` and `hooks:check` (self-test plus drift check against committed `docs/hooks.md`); Travis runs `yarn run hooks:check`. README links to the new reference. A PHP fixture under `tests/fixtures/` exercises parser edge cases (qualified calls, dotted literals, placeholders).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2c015febf1fec7398201fd9108fca3e2cb79e407. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: bin/generate-hooks-docs.php [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+			$value = \preg_replace( '/\$([A-Za-z_][A-Za-z0-9_]*)/', '{$1}', (string) $value );
+			$name .= \stripcslashes( (string) $value );
+		} elseif ( \preg_match( '/^\$([A-Za-z_][A-Za-z0-9_]*)(.*)$/s', $part, $matches ) ) {
+			$name .= '{' . normalize_placeholder( $matches[1] . $matches[2] ) . '}';
+		} else {
+			return null;
+		}
+	}
+
+	return $name;
```
[the remark this answers is not in the document]
Comment (jacobd91) [reply bin/generate-hooks-docs.php:385]: Fixed in 2c015feb. The generator now tokenizes concatenation operators outside quoted literals, preserves dotted hook names, and adds fixture coverage for quoted dotted names and concatenated expressions. The regenerated reference now includes the reported hooks; PHP 7.4 and PHP 8.5 self-test/check both pass.

**Review by cssjoe — APPROVED** [review]
Merge-ready on 2c015fe.

Isolated review of the full branch vs master: no medium/high issues. Generator `--self-test` and `--check` pass on PHP 7.4 and 8.2–8.5 (242 hooks; docs are current). PHPCS on `bin/generate-hooks-docs.php` is clean. Travis and GitHub checks on this SHA are green. The dotted-name parser fix is in the generated tables.

Optional follow-ups only (not blocking): `mkdir` mode 0777 when creating `docs/`; WP-cron hook names that are registered but never dispatched in production stay out of the scan-by-design reference.

Approving.
